hazelcast-jet-core/src/main/java/com/hazelcast/jet/Observable.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet;
+
+import com.hazelcast.jet.core.JetProperties;
+import com.hazelcast.jet.function.Observer;
+import com.hazelcast.jet.impl.observer.BlockingIteratorObserver;
+import com.hazelcast.ringbuffer.Ringbuffer;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Represents a flowing sequence of events produced by one or more {@link
+ * com.hazelcast.jet.pipeline.Sinks#observable(String) observable sinks}.
+ * To observe the events, call {@link #addObserver
+ * jet.getObservable(name).addObserver(myObserver)}.
+ * <p>
+ * The identity of an {@code Observable} is its name. You can get any
+ * number of {@code Observable} objects by the same name and they will
+ * all correspond to the same entity. All registered observers will get
+ * all the published events. Likewise, you can create any number of sinks
+ * using the same name and they will all push the events to the same
+ * {@code Observable}.
+ * <p>
+ * Even though an observable sink is the only way to publish data to an
+ * {@code Observable}, their lifecycles are decoupled. You create an
+ * {@code Observable} when you acquire it by name, either by running a
+ * job with an observable sink or by calling {@link JetInstance#getObservable
+ * jet.getObservable()}, and after that it stays alive until you explicitly
+ * {@link #destroy} it, or until the auto-cleanup mechanism kicks in and
+ * destroys it for you. (Keep in mind that a job may complete, but the
+ * {@code Observable}, along with the data it published to it, lives on.
+ * <strong>Consuming the data does not remove it from the
+ * Observable</strong>. Also, if you destroy an {@code Observable} that is
+ * still in active use by a sink, it will silently re-create it the next
+ * time it has data to push to it.)
+ * <p>
+ * Auto-cleanup of the observables is a timeout based mechanism. The timeout
```

Comment:
This explanation of the timeout mechanism is too complex. We should simplify the description to something easy to reason about, like:

```
 * Auto-cleanup of the observables is a timeout-based mechanism. The
 * timeout defaults to a week and can be configured via {@link
 * JetProperties#JOB_RESULTS_TTL_SECONDS}. It starts as soon as there are
 * no jobs running that contain a sink for a given observable.
```
If this is not how it actually works, we should simplify the design/implementation as well.